cmd/kubectl-gadget/utils/trace.go

```diff
@@ -637,6 +654,25 @@ func RunTraceAndPrintStatusOutput(config *TraceConfig, customResultsDisplay func
 	return PrintTraceOutputFromStatus(traceID, config.TraceOutputState, customResultsDisplay)
 }
 
+// SetTraceOpAndPrintStatusOutput sets an annotation with the requested
+// operation, prints the the Status.Output when the Status.State goes
+// in the expected one and, finally, it deletes the trace.
+// It is equivalent to call separately SetTraceOperation(), then
+// PrintTraceOutputFromStatus() and DeleteTrace().
+// This function is thought to be used by "multi-run" gadget.
+func SetTraceOpAndPrintStatusOutput(traceID, requestedOperation, expectedStatus string,
+	customResultsDisplay func(results *gadgetv1alpha1.TraceList) error,
+) error {
+	err := SetTraceOperation(traceID, requestedOperation)
+	if err != nil {
+		return err
+	}
+
+	defer DeleteTrace(traceID)
+
+	return PrintTraceOutputFromStatus(traceID, expectedStatus, customResultsDisplay)
+}
```

Comment:
It's not immediate to recognize that this function removes the trace from the name. If you look at the implementation of `runBiolatencyStop` it's not clear that the trace was removed there. Do we really need this kind of highly specialized functions to be part of the utils package? I think the user can use `SetTraceOperation`, `DeleteTrace` and `PrintTraceOutputFromStatus` directly. If we find later on that a lot of gadgets are calling these 3 functions in the same way we introduce a wrapper like this one.

Reply:
Yeah, I added this function because I expect that most of the multi-run gadgets will need to do the same, e.g. seccomp and network-policy-advisor. In addition, I was expecting that the function description was enough to guide developers. However, it seems to be not enough because, as you said, it is not that immediate if one looks at the `runBiolatencyStop` implementation.

Mmmm I think I will move this to the biolatency code and once we start adding the new CLIs we can think about adding it. Maybe using a name that makes it clearer that the trace is being also deleted.